metpy/cbook.py

```diff
@@ -21,16 +24,19 @@ def is_string_like(s):
     return isinstance(s, string_type)
 
 
-def get_test_data(fname, as_file_obj=True):
-    """Access a file from MetPy's collection of test data."""
-    # Look for an environment variable to point to the test data. If not, try looking at
-    # the appropriate path relative to this file.
-    data_dir = os.environ.get('TEST_DATA_DIR',
-                              os.path.join(os.path.dirname(__file__), '..', 'staticdata'))
+POOCH = pooch.create(
+    path=pooch.os_cache('metpy'),
+    base_url='https://github.com/Unidata/MetPy/raw/{version}/staticdata/',
+    version=get_versions()['version'],
+    version_dev='master',
+    env='TEST_DATA_DIR',
+)
+POOCH.load_registry(os.path.join(os.path.dirname(__file__), 'registry.txt'))
 
-    # Assemble the path
-    path = os.path.join(data_dir, fname)
 
+def get_test_data(fname, as_file_obj=True):
+    """Access a file from MetPy's collection of test data."""
+    path = POOCH.fetch(fname)
```

Comment:
Would be good to have a way to use our local git checkout of files directly so we can more easily bootstrap adding new files.

Reply:
I thought of that while implementing this. The problem is that the directory is versioned. I got around it on the CIs by copying the data to the cache folder. That bypasses the download at least. A static link would be better locally to keep the folders in sync.

Reply:
I think the way to do this is check "are we in a git repo?" and if so, then we'll set the environment variable to point the cache to the staticdata directory.

Reply:
That wouldn't work because of the versioning. The env variable only points to the base level directory.

Reply:
Just got a workaround going - verifying now.